blockchain/hash.go

```diff
@@ -26,3 +55,43 @@ type Marshaller interface {
 func HashSum(m Marshaller) Hash {
 	return sha256.Sum256(m.Marshal())
 }
+
+// CompareTo compares two hashes, it returns true if the operation of the first hash on the second hash specified by the comparator is true, and false otherwise
+func CompareTo(h1 Hash, h2 Hash, comparator int) bool {
```

Comment:
We will only ever be checking for hashes being less than right? Can we simplify this accordingly and maybe remove those comparator constants unless we need arbitrary comparisons?